### PR TITLE
Security event logging

### LINKED ISSUE #4577: Security event logging
Create a database table to store information on security-related events (failed & successful logins, password resets, email changes, token verification failures, etc.).

Related:
- #4087
- [Security Log plugin](https://community.mybb.com/mods.php?action=view&pid=740)

### LINKED ISSUE #4578: List logged security events in the ACP
Allow administrators to view logged security events in the Admin CP.

Depends on #4577

Related:
- [Security Log plugin](https://community.mybb.com/mods.php?action=view&pid=740)

### PR DESCRIPTION
**Pull request by Starpaul20**
Starting work on #4577

### CODE REVIEW

**Comment by dvz** [comment]
Tangentially related - some practical changes made to the Security Log plugin: https://github.com/Starpaul20/Security-Log/pull/3

File: usercp.php [hunk 1]
```diff
@@ -3332,6 +3344,91 @@
...
+		$logitem['ipaddress'] = my_inet_ntop($db->unescape_binary($logitem['ipaddress']));
+
+		$plugins->run_hooks('usercp_securitylog_result');
+
+		$securitylog[] = $logitem;
+	}
+
+	$plugins->run_hooks('usercp_securitylog_end');
+
+	output_page(\MyBB\template('usercp/securitylog.twig', [
```
Comment (Sama34) [inline usercp.php]:
```suggestion
	output_page(\MyBB\View\template('usercp/securitylog.twig', [
```

File: usercp.php [hunk 2]
```diff
@@ -3332,6 +3344,91 @@
...
+	{
+		$start = 0;
+		$page = 1;
+	}
+
+	if($postcount > $perpage)
+	{
+		$multipage = multipage($postcount, $perpage, $page, "usercp.php?action=securitylog");
+	}
+
```
Comment (Sama34) [inline usercp.php:3400]:
```suggestion

    $multipage = $multipage ?? '';
    
```

File: admin/modules/tools/securitylog.php [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+		$mybb->input['older_than'] = $mybb->get_input('older_than', MyBB::INPUT_INT);
+		if($mybb->input['older_than'] <= 0)
+		{
+			$is_today = true;
+			$mybb->input['older_than'] = 1;
+		}
+		$where = 'dateline < '.(TIME_NOW-($mybb->input['older_than']*86400));
+
+		// Searching for entries by a particular user
+		if($mybb->input['uid'])
```
Comment (Sama34) [inline admin/modules/tools/securitylog.php]:
```suggestion
		if($mybb->get_input('uid') !== '')
```

**Review by Sama34 — CHANGES_REQUESTED** [review]

**Review by Sama34 — APPROVED** [review]

**Pull request merged**